R/names.R

```diff
@@ -0,0 +1,405 @@
+#' Retrieve and repair names
+#'
+#' @description
+#'
+#' Like `base::names()`, `vec_names()` retrieves the names of a
+#' vector. It has the following differences:
+#'
+#' * It always returns __minimal names__. Whereas `names()` returns
+#'   `NULL` when a vector doesn't have names, `vec_names()` returns a
+#'   character vector of empty strings by default. `NA` names are also
+#'   converted to `""`. You can control how names are repaired with
+#'   the `repair` argument.
+#'
+#' * With arrays it returns the row names. With data frames it returns
+#'   an empty character vector as long as the number of rows.
+#'
+#' @param x A vector.
+#' @param ... These dots are for future extensions and must be empty.
+#' @param repair
+#'   * Minimal names are never `NULL` or `NA`. When an element doesn't
+#'     have a name, its minimal name is an empty string.
+#'
+#'   * Unique names are unique. A suffix is appended to duplicate
+#'     names to make them unique.
+#'
+#'   * Universal names are unique and syntactic, meaning that you can
+#'     safely use the names as variables without causing a syntax
+#'     error.
+#'
+#'   See [vec_repair_names()] for a complete overview of names repair.
+#' @param quiet By default, the user is informed of any renaming
+#'   caused by names repairs. This only concerns unique and universal
+#'   repairing. Set `quiet` to `TRUE` to silence the messages.
```

Comment:
Propose:

By default, the user is informed of any renaming caused by making names unique or universal.

Reply:
"caused by repairing the names"? Since the `repair` function might be supplied by the user.